swfiles/sw_fitpowder.m

```diff
@@ -476,6 +516,16 @@ function estimate_constant_background(obj)
             obj.set_bg_parameters(obj.nparams_bg, bg);  % set constant background
         end
 
+        function set_errors_of_bg_bins(obj, val)
+            if isempty(obj.ibg)
+                obj.find_indices_and_mean_of_bg_bins();
+            end
+            if nargin < 2
+                val = max(obj.e(obj.ibg));
+            end
+            obj.e(obj.ibg) = val;
```

Comment:
Is it worthwhile to keep a copy of the original data and errors in case we have to "undo" this operation?

I have a bit of a concern about the sequence of operations - e.g. if for a series of cuts we add some cuts, fit or estimate the background, call this function (`set_errors_of_bg_bins`), then add a new cut and call `estimate_constant_background` again - the new estimate of the constant background is be erroneous because of the background pixels set to a (larger) `val` here. (For this particular edge case, we could handle it with a private flag indicating of the background bins have been overwritten... but there might be other edge cases...)

Reply:
Here only the error of the background bins are overwritten not the value, and when more cuts are added `clear_cache` is called which sets `ibg=[]`. So calling `estimate_constant_background` should still work in this instance. However I can store the old error values and add an option to restore the original values (and do this automatically in `clear_cache`)?

Reply:
Yes, I think that this might be best (store the old errors and then restore them in `clear_cache`).